### DISCUSSION TITLE
docs: NestJS integration- question about RequestContext

### DISCUSSION THREAD

**Discussion by vladkrasn** [mikro-orm/mikro-orm#5297] [General]
Official documentation (https://mikro-orm.io/docs/usage-with-nestjs#request-scoped-handlers-in-queues) suggests "That is handled automatically thanks to the RequestContext helper registered via middleware."

What does it mean? It works out of the box and there's no need to register anything in middleware manually?
Certainly no middleware is registered in the example project https://github.com/mikro-orm/nestjs-example-app/

But when I first read it, I though that I need to register it explicitly like this
![image](https://github.com/mikro-orm/mikro-orm/assets/28753668/7f06ceb9-4ac2-4886-8904-e4ed101e0f7d)

What is the correct way?

**Comment by B4nan** [comment]
the linked section is (as its heading clearly states) about using the ORM inside queues, and there are no middlewares in queues. it feels like you read the first sentence and not the second one, which follow saying "But middlewares are executed only for regular HTTP request handles, what if we need a request scoped method outside of that? One example of that is queue handlers or scheduled tasks.".

**Reply by vladkrasn** [reply]
I got the part where it says that middleware doesn't work in non-HTTP context, and there are workarounds available. 
But I didn't get the part about whether I need to explicitly register middleware in HTTP context.

So, if I make a simple HTTP application (like the official example), there's no need to register any such middlewares?

**Reply by B4nan** [reply]
You don't need to register any middleware in nest, the `@mikro-orm/nestjs` adapter does that for you automatically. This part of the docs is about understanding that 1. there is a middleware that does this for you, and 2. some actions like queues are not executed in a middleware context, so you need other ways to bring the context in - one being the decorators, the other being manual forking. that first sentence links to https://mikro-orm.io/docs/identity-map which describes how those things work and why they are needed.

**Reply by vladkrasn** [reply]
Great! Thank you for your help!
